docs/spec/sources.md

```diff
@@ -362,5 +375,38 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects). It
+is highly-recommended to produce the `knsource` CloudEvent extension.
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+
+Examples:
+
+- `hello.pingsources.sources.knative.dev`
+- `rabbitmq-source.rabbitmq.sources.knative.dev`
+
+It is allowed to specify an alternative value in
+[`spec.ceOverrides.extensions`](#duckspec), in which case the `knsource` value
+can be of any shape.
+
+When the `knsource` CloudEvent extension already exists in the CloudEvent being
+forwarded by a Source, its value SHOULD be overridden as described above.
+
+One use case addressed by the `knsource` CloudEvent extension is event filtering
+based on [Source Custom Object](#source-custom-objects). For instance triggers
+can be configured to receive events coming from a single
+[Source Custom Object](#source-custom-objects).
+
```

Comment:
Do we have more than one use-case ?

Not really sure what this is trying to solve.

/assign @slinkydeveloper 
/assing @n3wscott

Reply:
Here the full context: https://github.com/knative/eventing/issues/1381

TL;TR: `One of the key motivations of the work around the origin of this issue is to support the FaaS scenario and make it possible to write side-effect free triggers.`.

Hopefully I'm not opening a can of worms by referring this long discussion. 😨